src/Loggers/Loggers.cpp

```diff
@@ -251,6 +267,15 @@ void Loggers::buildLoggers(Poco::Util::AbstractConfiguration & config, Poco::Log
                 {
                     const std::string name(config.getString("logger.levels." + key + ".name"));
                     const std::string level(config.getString("logger.levels." + key + ".level"));
+
+                    std::string pos_pattern = config.getRawString("logger.levels." + key + "message_regexp", "");
+                    std::string neg_pattern = config.getRawString("logger.levels." + key + "message_regexp_negative", "");
+
+                    if (auto * regexp_channel = dynamic_cast<DB::OwnFilteringChannel*>(logger.root().get(name).getChannel()))
+                        regexp_channel->setRegexpPatterns(pos_pattern, neg_pattern);
+                    else
+                        throw DB::Exception(DB::ErrorCodes::TYPE_MISMATCH, "Couldn't convert to OwnFilteringChannel.");
```

Comment:
`TYPE_MISMATCH` is an error code for type mismatch in the data, not in C++ types. I guess this shouldn't happen unless something is wrong, so feel free to use `LOGICAL_ERROR`.